lib/datastore.js

```diff
@@ -80,10 +80,37 @@ Datastore.prototype.loadDatabase = function () {
 
 
 /**
+ * Delete any documents from the array of documents that are expired
+ */
+Datastore.prototype._removeExpiredDocuments = function (docs) {
+  var self = this
+    , expiredDocs = []
+    , validDocs = [];
+
+  docs.forEach(function(doc) {
+    if (model.isExpired(doc) || doc.$$deleted === true) {
```

Comment:
@louischatriot: Is needing to check for ` || doc.$$deleted === true` at this point a realistic requirement, or does that field only exist on documents that are written to the disk for later compaction?  I wasn't sure so I erred on the side of caution.